contracts/socket/andromeda-socket-astroport/src/contract.rs

```diff
@@ -396,22 +347,47 @@ fn provide_liquidity(
                     expires: None,
                 };
                 let allowance_wasm_msg = wasm_execute(contract_addr, &allowance_msg, vec![])?;
-                response = response.add_message(allowance_wasm_msg);
+                response_msg.push(allowance_wasm_msg);
             }
         }
     }
 
     // Send the provide liquidity message to the pair (native coins attached, CW20s via allowance)
     let provide_wasm_msg = wasm_execute(pair_addr_raw, &provide_liquidity_msg, native_coins)?;
-    response = response.add_message(provide_wasm_msg);
+    response_msg.push(provide_wasm_msg);
+
+    let mut response: Response = Response::new();
+    for msg in response_msg {
+        response = response.add_message(msg);
+    }
 
     Ok(response.add_attributes(vec![
         attr("action", "provide_liquidity"),
-        attr("pair_address", pair_addr.to_string()),
+        attr("pair_address", pair_address.to_string()),
         attr("assets", format!("{:?}", assets)),
     ]))
 }
 
+fn execute_update_swap_router(
+    ctx: ExecuteContext,
+    swap_router: AndrAddr,
+) -> Result<Response, ContractError> {
+    let ExecuteContext { deps, info, .. } = ctx;
+
+    // Verify sender has owner permissions
+    ADOContract::default().is_contract_owner(deps.storage, info.sender.as_str())?;
```

Comment:
No need for this since we already have `#[attrs(restricted)]` for `UpdateSwapRouter`